### DISCUSSION TITLE
Fatal error when trying to generate workflow from local yaml spec

### DISCUSSION THREAD

**Discussion by designvoid** [stepci/stepci#208] [Q&A] [answered]
I am trying to generate a workflow from my spec and get the following error:
```
TypeError: Cannot read properties of undefined (reading 'default')
    at /Users/xxx/.nvm/versions/node/v18.10.0/lib/node_modules/stepci/node_modules/@stepci/plugin-openapi/index.js:74:26
    at Array.forEach (<anonymous>)
    at generateWorkflow (/Users/xxx/.nvm/versions/node/v18.10.0/lib/node_modules/stepci/node_modules/@stepci/plugin-openapi/index.js:72:123)
    at async generateWorkflowFile (/Users/xxx/.nvm/versions/node/v18.10.0/lib/node_modules/stepci/node_modules/@stepci/plugin-openapi/index.js:208:45)
    at async Object.handler (/Users/xxx/.nvm/versions/node/v18.10.0/lib/node_modules/stepci/dist/index.js:145:5)
```

I have attached my spec - compressed as it's quite large.

Any help would be supper appreciated! Thanks!!!

**Comment by mishushakov** [comment]
Fixed in the openapi plugin 0.4.1.

Btw. your spec is partially faulty, you should use a tool like optic (useptic.com) or swagger editor (editor-next.swagger.io) to fix it.
Thanks!

**Reply by designvoid** [reply]
Hey @mishushakov how do I update the openapi plugin to check that?

**Reply by mishushakov** [reply]
Reinstall the CLI (no cache)

**Reply by designvoid** [reply]
Sorry - I have totally uninstalled the package and reinstalled but still seeing the same issues :(

**Reply by designvoid** [reply]
Additionally - I have validated my Spec with Swagger hub and it's all green?

**Reply by mishushakov** [reply]
Ok, I have updated the fix. Try again. Anyways I'm getting lots of errors with your spec

<img width="877" alt="Screenshot 2024-03-01 at 19 39 38" src="https://github.com/stepci/stepci/assets/10400064/06c38bc0-37a1-4330-bc5e-eae381f2caf2">

**Reply by designvoid** [reply]
Hey @mishushakov it looks like it doesn't like the fact the spec is OpenAPI 3.1.0?

I am using Stoplight and the `JSON:API` validator and the built in OpenAPI one and I don't see any of these errors? :(

**Reply by designvoid** [reply]
It seems to be having an issue with the new way of describing nullable properties:
```yaml
                        description:
                          type:
                            - string
                            - 'null'
```

This is valid for OpenAPI 3.1 afaik?

**Reply by designvoid** [reply]
any thoughts on this @mishushakov ?

**Reply by mishushakov** [reply, marked as answer]
We don't support OpenAPI 3.1

**Reply by designvoid** [reply]
But your documentation states

<img width="745" alt="image" src="https://github.com/stepci/stepci/assets/630948/5523b712-9102-40a3-b29f-341b3e51d17f">

**Reply by mishushakov** [reply]
Depends on the spec. Some specs are just too complex to digest and tbh, the OpenAPI standard is a bit messy in some regards.
